lib/chef/provider/git.rb

```diff
@@ -34,99 +34,55 @@ def whyrun_supported?
       def load_current_resource
         @resolved_reference = nil
         @current_resource = Chef::Resource::Git.new(@new_resource.name)
-        if current_revision = find_current_revision
-          @current_resource.revision current_revision
-        end
+
+        current_revision = find_current_revision
+        @current_resource.revision current_revision if current_revision
```

Comment:
i actually like the previous form of this better, i don't think there's any edge conditions that it improves.